### ISSUE TITLE
better management of gh-actions jobs

### ISSUE THREAD

**Issue by elstoc** [darktable-org/darktable#11106]
At the moment we have quite a lot of long running github actions jobs and because we can only use a limited number of runners at any one time, they are often queued for ages (for example as I write this issue one CI job has been queued for over 5 hours). Some of this can perhaps be managed with the [`concurrency`](https://docs.github.com/en/actions/using-jobs/using-concurrency) option, but there may be other options I'm not aware of. Some thoughts anyway:

- Sometimes a PR is merged before its CI has run. The act of merging to master then triggers another CI run. It might be nice to be able to cancel PR CI runs if they have been subsequently merged. Or don't merge PRs until the CI has run successfully.
- Given that people push to PRs quite often it might also be good if we could limit so that only one PR CI can run at a time, thus leaving runners available for other types of job. The concurrency option would probably do this.
- Sometimes lots of PRs are merged to master in quick succession. Each merge triggers a new CI. Do we need all of these to run on master? Again, concurrency options could ensure that these only run one at a time, or we could cancel any previous runs if another merge is completed (using `cancel-in-progress: true`)
- We have an action that posts a list of recent commits to the darktable IRC channel. This runs fairly frequently but due to the other jobs using all the runners (for ages), we often have a number of queued IRC bot jobs (7 queued jobs right now). Once a runner becomes available, sometimes more than one runs concurrently. It would be nice to be able to cancel all queued jobs when one of them actually runs. I'm not sure that `cancel-in-progress` is the right option for this as we probably would prefer to cancel the _current_ run instead of any already executing. Ensuring that similar jobs can't run concurrently would probably help though as sometimes two concurrent runs can interfere with one another.

**Comment by elstoc** [comment]
So based on #11107 it seems that the "concurrency" option usually leaves at least two jobs in the queue. So, if three jobs are set up one after another, without any of them being triggered, it works as follows:

- Run 1: job 1 is queued
- Run 2: since job 1 is already queued, job 2 is placed into pending status
- Run 3: job 2 is still pending and so job 2 is cancelled and superseded by job 3

So that fix means that two IRC notification jobs can't run at the same time and keeps the queue to 2 jobs max, but could still mean that two jobs run immediately after each other. Which IMO is at least better than it was.
